tests/BuildScriptGenerator.Tests/Node/NodeBuildScriptGenerationTest.cs

```diff
@@ -123,6 +123,7 @@ public void GeneratedScript_HasNpmVersion_SpecifiedInPackageJson()
                     {"PlatformWithVersion", "Node.js 10.10.10" },
                 },
                 NodeBuildCommandsFile = FilePaths.BuildCommandsFileName,
+                NpmVersionSpec = "5.4.2",
```

Comment:
Would we be able to test with a valid and invalid `NpmVersionSpec` and check that `5.4.2` was selected for the build

Reply:
This might be a bit tricky when the user specifies something like this `~5.4.2` or other loose version specification

Reply:
Per our discussion offline: we grab the entire version spec that the user provided in the `package.json` file and use that in our `npm install -g npm@<spec>` command.